### PR TITLE
Call Ibex using codegen sdk with authentication handling

### PR DESCRIPTION
**Pull request by brh28**
- Wraps the Ibex codegen sdk to with authentication and logging handling, and maps errors to application errors of type IbexEventError. 
- Application logic (primarily found in graphQL resolvers) modified to call newly created Ibex service with minimal changes.  
- Also contains some documentation updates and example queries to Ibex.

### CODE REVIEW

**Comment by brh28** [comment]
Code generation is checked into source control under "src/services/ibex/.api/". This would probably be better off taken out of source control and added to the build, but shouldn't be a problem for now.

File: src/graphql/public/root/mutation/onchain-usd-payment-send.ts [hunk 1]
```diff
@@ -79,48 +78,49 @@ const OnChainUsdPaymentSendMutation = GT.Field<
...
-              ? "failed"
-              : "pending",
-        },
-        payoutId: PayOnChainAddress.data["data"]["transactionHub"]["id"],
-      }
 
-      if (result instanceof Error) {
-        return { status: "failed", errors: [mapAndParseErrorForGqlResponse(result)] }
-      }
+    const resp = await Ibex.sendToAddressV2({
```
Comment (brh28) [inline src/graphql/public/root/mutation/onchain-usd-payment-send.ts:82]: Need to test the status field returned here. Previous code seems to be mapped

File: src/app/wallets/create-on-chain-address.ts [hunk 2]
```diff
@@ -1,87 +1,114 @@
 import { AccountValidator } from "@domain/accounts"
```
Comment (brh28) [inline src/app/wallets/create-on-chain-address.ts:1]: Retest file. Clean commented code

**Review by islandbitcoin — APPROVED** [review]
ACK - building successful

File: src/services/ibex/index.ts [hunk 3]
```diff
@@ -0,0 +1,78 @@
+// API registry: https://dash.readme.com/api/v1/api-registry/cpd51bloegfhl2
+import IbexSDK, { AddInvoiceBodyParam, AddInvoiceResponse201, CreateAccountBodyParam, CreateAccountResponse201, EstimateFeeCopyMetadataParam, EstimateFeeCopyResponse200, GMetadataParam, GResponse200, GResponse400, GenerateBitcoinAddressBodyParam, GenerateBitcoinAddressResponse201, GetAccountDetailsMetadataParam, GetAccountDetailsResponse200, GetFeeEstimationMetadataParam, GetFeeEstimationResponse200, PayInvoiceV2BodyParam, PayInvoiceV2Response200, RefreshAccessTokenBodyParam, SendToAddressCopyBodyParam, SendToAddressCopyResponse200, SignInResponse200 } from "./.api/apis/sing-in" // TODO: @sing-in@<uuid>
```
Comment (brh28) [inline src/services/ibex/index.ts:2]: change to `import IbexSDK, *`

**Comment by islandbitcoin** [comment]
Integration testing review:

- I know we agreed that the {me,defaultAccount,wallets,0,balance} GraphQL query doesn't need to work for the BTC side, but we should probably leave it working for future interoperability. Let's make a TODO for that if we cannot fix it in this PR. For the demo can we uncomment your 'hack' that sets the unknown walletIds to zero?

**Comment by islandbitcoin** [comment]
> Integration testing review:
> 
> * I know we agreed that the {me,defaultAccount,wallets,0,balance} GraphQL query doesn't need to work for the BTC side, but we should probably leave it working for future interoperability. Let's make a TODO for that if we cannot fix it in this PR. For the demo can we uncomment your 'hack' that sets the unknown walletIds to zero?

Second thing I noticed - there is no webhook, or any response to let the frontend know that a payment has been received or sent. That was originally coming from galoy, I'm assuming we would need to now use a webhook from Ibex

**Comment by islandbitcoin** [comment]
> > Integration testing review:
> > 
> > * I know we agreed that the {me,defaultAccount,wallets,0,balance} GraphQL query doesn't need to work for the BTC side, but we should probably leave it working for future interoperability. Let's make a TODO for that if we cannot fix it in this PR. For the demo can we uncomment your 'hack' that sets the unknown walletIds to zero?
> 
> Second thing I noticed - there is no webhook, or any response to let the frontend know that a payment has been received or sent. That was originally coming from galoy, I'm assuming we would need to now use a webhook from Ibex

third thing - the backend seems to be creating two accounts on IBEX when creating a new wallet. Probably because it's running the function for both BTC and USD wallets. That's causing some confusion on the IBEX side.

File: src/services/mongoose/wallets.ts [hunk 4]
```diff
@@ -36,28 +35,12 @@ export const WalletsRepository = (): IWalletsRepository => {
...
-          !IbexAccountCreationResponse ||
-          !IbexAccountCreationResponse.data ||
-          !IbexAccountCreationResponse.data["data"]["id"]
-        ) {
-          console.error({ error: "unable to get IbexAccountCreationResponse" })
-        } else {
-          ibexAccountId = IbexAccountCreationResponse.data["data"]["id"]
-        }
-      }
+      const resp = await Ibex.createAccount({
```
Comment (islandbitcoin) [inline src/services/mongoose/wallets.ts:38]: Does this need an if statement or something so it only makes a new account on IBEX when the currency === 'USD' ?

Comment (islandbitcoin) [reply src/services/mongoose/wallets.ts:38]: this seems to work
```

let ibexAccountId
    try {
      // FLASH FORK: create IBEX account if currency is USD
      if (currency === "USD") {
        const resp = await Ibex.createAccount({
          name: accountId,
          currencyId: 3,
        })
        if (resp instanceof IbexEventError) return resp
        ibexAccountId = resp.id
      }
      const wallet = new Wallet({
        _accountId: toObjectId<AccountId>(accountId),
        id: ibexAccountId || crypto.randomUUID(),
```

**Comment by islandbitcoin** [comment]
last test result for the night: lnNoAmountUsdInvoiceFeeProbe.

The GraphQL query is failing
Screenshot 2024-01-08 at 12 56 10 AM

File: src/app/wallets/create-on-chain-address.ts (hunk 2 above)
Comment (brh28) [reply src/app/wallets/create-on-chain-address.ts:1]: onchain address created. additional testing may be required

**Pull request merged**